validator/main.go

```diff
@@ -614,24 +664,67 @@ func assertDriverContainerReady(silent bool) error {
 	return runCommand(command, args, silent)
 }
 
-func (d *Driver) runValidation(silent bool) (string, bool, error) {
-	driverRoot, isHostDriver := getDriverRoot()
-	if !isHostDriver {
-		log.Infof("Driver is not pre-installed on the host. Checking driver container status.")
+func validateHostDriver(silent bool) error {
+	log.Info("Attempting to validate a pre-installed driver on the host")
+	command := "chroot"
+	args := []string{"/host", "nvidia-smi"}
+
+	return runCommand(command, args, silent)
+}
+
+func validateDriverContainer(silent bool) error {
+	if driverManagedByOperator {
+		log.Infof("Driver is not pre-installed on the host and is managed by GPU Operator. Checking driver container status.")
 		if err := assertDriverContainerReady(silent); err != nil {
-			return "", false, fmt.Errorf("error checking driver container status: %v", err)
+			return fmt.Errorf("error checking driver container status: %w", err)
 		}
 	}
 
-	// invoke validation command
-	command := "chroot"
-	args := []string{driverRoot, "nvidia-smi"}
+	driverRoot := root(driverRootFlag)
 
-	if withWaitFlag {
-		return driverRoot, isHostDriver, runCommandWithWait(command, args, sleepIntervalSecondsFlag, silent)
+	validateDriver := func(silent bool) error {
+		driverLibraryPath, err := driverRoot.getDriverLibraryPath()
+		if err != nil {
+			return fmt.Errorf("failed to locate driver libraries: %w", err)
+		}
+
+		nvidiaSMIPath, err := driverRoot.getNvidiaSMIPath()
+		if err != nil {
+			return fmt.Errorf("failed to locate nvidia-smi: %w", err)
+		}
+		cmd := exec.Command(nvidiaSMIPath)
+		// In order for nvidia-smi to run, we need to update LD_PRELOAD to include the path to libnvidia-ml.so.1.
+		cmd.Env = setOrOverrideEnvvar(os.Environ(), "LD_PRELOAD", prependPathListEnvvar("LD_PRELOAD", driverLibraryPath))
+		if !silent {
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+		}
+		return cmd.Run()
```

Comment:
This is not running in a `chroot`. Does this mean that we will end up creating the device nodes at `/dev` in the container. Is the `devPath` ever applicable in this case?

Reply:
My understanding is that `nvidia-smi` expects the device nodes created at `/dev` and that there isn't a way to control this.